Tests/test_file_avif.py

```diff
@@ -583,22 +550,20 @@ def test_decoder_codec_available_invalid(self) -> None:
         assert _avif.decoder_codec_available("foo") is False
 
     @pytest.mark.parametrize("upsampling", ["fastest", "best", "nearest", "bilinear"])
-    def test_decoder_upsampling(self, upsampling: str) -> None:
-        AvifImagePlugin.CHROMA_UPSAMPLING = upsampling
-        try:
+    def test_decoder_upsampling(
+        self, monkeypatch: pytest.MonkeyPatch, upsampling: str
+    ) -> None:
+        monkeypatch.setattr(AvifImagePlugin, "CHROMA_UPSAMPLING", upsampling)
+
+        with Image.open(TEST_AVIF_FILE):
+            pass
+
+    def test_decoder_upsampling_invalid(self, monkeypatch: pytest.MonkeyPatch) -> None:
+        monkeypatch.setattr(AvifImagePlugin, "CHROMA_UPSAMPLING", "foo")
+
+        with pytest.raises(ValueError):
             with Image.open(TEST_AVIF_FILE):
                 pass
-        finally:
-            AvifImagePlugin.CHROMA_UPSAMPLING = "auto"
-
-    def test_decoder_upsampling_invalid(self) -> None:
-        AvifImagePlugin.CHROMA_UPSAMPLING = "foo"
-        try:
-            with pytest.raises(ValueError):
-                with Image.open(TEST_AVIF_FILE):
-                    pass
-        finally:
-            AvifImagePlugin.CHROMA_UPSAMPLING = "auto"
```

Comment:
I accidentally omitted one line in `_avif.c`, which should read:

```c
self->decoder->chromaSampling = upsampling;
```

I think I would rather add the missing piece of the implementation, in which case I think it's worth keeping this test so that there is coverage for the validation of this parameter.

Reply:
I think you have been confused by the diff - I'm not removing this test. I'm just changing it to use monkeypatch.

Reply:
You're right, my mistake.